packages/database/src/migrations/20200914013941-AddTextbookStudentRatioOverlaysDistrictAndProvinceLevel-modifies-data.js

```diff
@@ -0,0 +1,631 @@
+    },
+    // By grade
+    {
+      numeratorCodes: ['STCL005'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 6'],
+      name: 'Mathematics - Grade 6',
+      baseId: 'Laos_Schools_Mathematics_Textbooks_Student_Ratio_Grade_6',
+    },
+    {
+      numeratorCodes: ['STCL024'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 7'],
+      name: 'Mathematics - Grade 7',
+      baseId: 'Laos_Schools_Mathematics_Textbooks_Student_Ratio_Grade_7',
+    },
+    {
+      numeratorCodes: ['STCL043'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 8'],
+      name: 'Mathematics - Grade 8',
+      baseId: 'Laos_Schools_Mathematics_Textbooks_Student_Ratio_Grade_8',
+    },
+    {
+      numeratorCodes: ['STCL062'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 9'],
+      name: 'Mathematics - Grade 9',
+      baseId: 'Laos_Schools_Mathematics_Textbooks_Student_Ratio_Grade_9',
+    },
+
+    // Natural Science
+    // Average
+    {
+      numeratorCodes: ['STCL006', 'STCL025', 'STCL044', 'STCL063'],
+      denominatorCodes: POP_DATA_ELEMENTS['Lower Sec'],
+      name: 'Natural Science - Average lower secondary',
+      baseId: 'Laos_Schools_Natural_Science_Textbooks_Student_Ratio_Lower_Secondary',
+    },
+    // By grade
+    {
+      numeratorCodes: ['STCL006'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 6'],
+      name: 'Natural Science - Grade 6',
+      baseId: 'Laos_Schools_Natural_Science_Textbooks_Student_Ratio_Grade_6',
+    },
+    {
+      numeratorCodes: ['STCL025'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 7'],
+      name: 'Natural Science - Grade 7',
+      baseId: 'Laos_Schools_Natural_Science_Textbooks_Student_Ratio_Grade_7',
+    },
+    {
+      numeratorCodes: ['STCL044'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 8'],
+      name: 'Natural Science - Grade 8',
+      baseId: 'Laos_Schools_Natural_Science_Textbooks_Student_Ratio_Grade_8',
+    },
+    {
+      numeratorCodes: ['STCL063'],
+      denominatorCodes: POP_DATA_ELEMENTS['Grade 9'],
+      name: 'Natural Science - Grade 9',
+      baseId: 'Laos_Schools_Natural_Science_Textbooks_Student_Ratio_Grade_9',
+    },
+
+    // Upper secondary
+    // (Only Average)
+    {
+      numeratorCodes: ['STCL080', 'STCL096', 'STCL112'],
+      denominatorCodes: POP_DATA_ELEMENTS['Upper Sec'],
+      name: 'Lao Language & Literature - Average upper secondary',
+      baseId: 'Laos_Schools_Lao_Language_Textbooks_Student_Ratio_Upper_Secondary',
+    },
+    {
+      numeratorCodes: ['STCL081', 'STCL097', 'STCL113'],
+      denominatorCodes: POP_DATA_ELEMENTS['Upper Sec'],
+      name: 'Mathematics - Average upper secondary',
+      baseId: 'Laos_Schools_Mathematics_Textbooks_Student_Ratio_Upper_Secondary',
+    },
+  ],
+};
+
+const mapToLevel = (overlays, level) =>
+  overlays.map(overlayConfig => ({
+    ...overlayConfig,
+    measureLevel: level,
+    id: `${overlayConfig.baseId}_${level}`,
+  }));
+
+const OVERLAYS_IN_HIERARCHY = {
```

Comment:
Note that this structure is not what was asked for in the card and may need to be altered.

Reply:
Has been approved in the card. Resolving 🎉